spec/models/application_spec.rb

```diff
@@ -75,6 +75,13 @@
       subject.send(:"#{flag}=", '0')
       expect(subject.flags).not_to include(flag)
     end
+
+    it 'ensures flags are unique' do
+      flag = flags.sample.to_s
+      subject.flags = [flag, flag]
+      subject.save!
```

Comment:
Just a suggestion: maybe we can save some microseconds by not persisting the record? Something like:
```ruby
it 'ensures flags are unique' do
  subject.flags = %w(fun fun)
  expect { subject.validate }.to change(subject, :flags).from(%w(fun fun)).to(%w(fun))
end
```
But no biggie, the way it is put right now is also perfectly fine.

Reply:
Makes sense to me, good call!